src/components/views/messages/DecryptionFailureBody.tsx

```diff
@@ -14,23 +14,37 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import React, { forwardRef, ForwardRefExoticComponent } from "react";
+import React, { forwardRef, ForwardRefExoticComponent, useContext } from "react";
 import { MatrixEvent } from "matrix-js-sdk/src/matrix";
+import { DecryptionFailureCode } from "matrix-js-sdk/src/crypto-api";
 
 import { _t } from "../../../languageHandler";
 import { IBodyProps } from "./IBodyProps";
+import { LocalDeviceVerificationStateContext } from "../../../contexts/LocalDeviceVerificationStateContext";
 
-function getErrorMessage(mxEvent?: MatrixEvent): string {
-    return mxEvent?.isEncryptedDisabledForUnverifiedDevices
-        ? _t("timeline|decryption_failure|blocked")
-        : _t("timeline|decryption_failure|unable_to_decrypt");
+function getErrorMessage(mxEvent: MatrixEvent, isVerified: boolean | undefined): string {
+    if (mxEvent.isEncryptedDisabledForUnverifiedDevices) return _t("timeline|decryption_failure|blocked");
+    switch (mxEvent.decryptionFailureReason) {
+        case DecryptionFailureCode.HISTORICAL_MESSAGE_NO_KEY_BACKUP:
+            return _t("timeline|decryption_failure|historical_event_no_key_backup");
+
+        case DecryptionFailureCode.HISTORICAL_MESSAGE_BACKUP_UNCONFIGURED:
+        case DecryptionFailureCode.HISTORICAL_MESSAGE_WORKING_BACKUP:
+            if (isVerified === false) {
```

Comment:
I think a comment here would be helpful here. Maybe something like

```suggestion
            // The user seems to have a working key backup, so prompt them to
            // verify in the hopes that when we restore from backup, we'll get
            // the key for this message.
            if (isVerified === false) {
```

Reply:
Hrm, this made me realise that we shouldn't be using this path for `HISTORICAL_MESSAGE_WORKING_BACKUP`